arduino-ide-extension/src/browser/dialogs/ide-updater/ide-updater-component.tsx

```diff
@@ -0,0 +1,139 @@
+import { nls } from '@theia/core/lib/common';
+import {} from '@theia/core/shared/react';
+import { shell } from 'electron';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import ReactMarkdown from 'react-markdown';
+import { ProgressInfo } from '../../../common/protocol/ide-updater-service';
+import ProgressBar from '../../components/ProgressBar';
+import { UpdateInfo } from './ide-updater-dialog';
+
+export type IDEUpdaterComponentProps = {
+  updateInfo: UpdateInfo;
+  downloadFinished?: boolean;
+  downloadStarted?: boolean;
+  progress?: ProgressInfo;
+  error?: Error;
+  onDownload: () => void;
+  onClose: () => void;
+  onSkipVersion: () => void;
+  onCloseAndInstall: () => void;
+};
+
+export const IDEUpdaterComponent = ({
+  updateInfo: { version, changelog },
+  downloadStarted = false,
+  downloadFinished = false,
+  progress,
+  error,
+  onDownload,
+  onClose,
+  onSkipVersion,
+  onCloseAndInstall,
+}: IDEUpdaterComponentProps): React.ReactElement => {
+  const changelogDivRef = React.useRef() as React.MutableRefObject<
+    HTMLDivElement
+  >;
+  React.useEffect(() => {
+    if (!!changelog) {
+      ReactDOM.render(
+        <ReactMarkdown
+          components={{
+            a: ({ href, children, ...props }) => (
+              <a onClick={() => href && shell.openExternal(href)} {...props}>
+                {children}
+              </a>
+            ),
+          }}
+        >
+          {changelog}
+        </ReactMarkdown>,
+        changelogDivRef.current
+      );
+    }
+  }, [changelog]);
+  const closeButton = (
+    <button onClick={onClose} type="button" className="theia-button secondary">
+      {nls.localize('arduino/ide-updater/notNowButton', 'Not now')}
+    </button>
+  );
+
+  return (
+    <div className="ide-updater-dialog--content">
+      {downloadFinished ? (
+        <div className="ide-updater-dialog--downloaded">
+          <div>Arduino IDE {version} has been downloaded.</div>
+          <div>Close the software and install the update on your machine.</div>
```

Comment:
```suggestion
          <div>
            {nls.localize(
              'arduino/ide-updater/versionDownloaded',
              'Arduino IDE {0} has been downloaded.',
              version
            )}
          </div>
          <div>
            {nls.localize(
              'arduino/ide-updater/closeToInstallNotice',
              'Close the software and install the update on your machine.'
            )}
          </div>
```

Not sure the ids are the best in this case.